src/Utils.js

```diff
@@ -19,6 +19,10 @@ export function internalInstance(inst) {
     inst[internalInstanceKey(inst)];
 }
 
+export function isFunctionalComponent(inst) {
+  return inst.constructor.name === 'StatelessComponent';
```

Comment:
let's maybe fence this with `inst && inst.constructor &&` anyway? I intend on using this in several places, and it seems possible for us to pass null in at some point...